docs/RFCS/txn_gc.md

```diff
@@ -0,0 +1,62 @@
+- Feature Name: txn-gc
+- Status: draft
+- Start Date: 2015-11-11
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+* rename `{Response->Sequence}Cache`.
+* store client's intents in `Txn` record on (successful) `EndTransaction`.
+* GC transaction records asynchronously following the client's `EndTransaction` call after successful resolution of (possibly) outstanding intents.
+* GC sequence cache entries with `ResolveIntent{,Range}` which are carried out as a consequence of client's `EndTransaction` (successful or not).
+* Check the sequence cache on non-writes as well and store the epoch along with the sequence (to allow for the next point):
+* Poison the sequence cache when aborting an intent after `Push` during `ResolveIntent{,Range}`, preventing the anomaly in #2231 (own writes vanishing).
+* Keep the "slow" gc queue which cleans up in case of abandoned txns or node crashes.
+
+# Motivation
+
+Both transaction and sequence cache records should be deleted when they're no longer useful, ideally without introducing extra work. The procedure outlined here accomplishes that in the vast majority of cases (including all non-abandoned transactions).
+
+# Detailed design
+
+Refresher:
+
+* there is exactly one txn record on a single range (the one that the
+  transaction is initiated on via `BeginTransaction`).
+* there is exactly one sequence cache entry for each `Range` mutated by the txn
```

Comment:
Actually not unrelated, especially the GC story has a caveat. On both `Split` and `Merge` we'll copy the entry (keeping the larger one on collision).

The GC story becomes a tiny bit more complicated here: entries will end up on a `Range` which doesn't necessarily get hit by a `ResolveIntent{,Range}` of the transaction (on a `Split`). These entries will still have to be GCed at some point (via the slow queue). Two issues with that:

* we need `Txn.Key` to get the range which holds the transaction entry, but we only have `Txn.ID` (from the sequence table entry).
* we need `Txn.Timestamp` to figure out whether this is "probably" an inactive transaction, but we don't have that either.

I'm not crazy about adding all of those bytes in there, but I don't see another option to get the slow queue working (also for abandoned transactions).

I'm going to add this to the document, but let me know if you have any other suggestions.